### PR TITLE
PGOV-809 Edge cases impacting document file versus efiles validation …

### PR DESCRIPTION
**Pull request by kxwang**
…and file size/mime type calculations

### CODE REVIEW

File: web/modules/custom/portland/portland.module [hunk 1]
```diff
@@ -824,6 +824,17 @@ function _validate_document(&$form, \Drupal\Core\Form\FormStateInterface $form_s
       $efiles_link = $form_state->getValue('field_efiles_link');
       $document = $form_state->getValue('field_document');
     }
+
+    // PGOV-809 Validate document and efiles link fields
+    if( ( empty($efiles_link[0]["uri"]) && empty($document[0]["fids"]) ) ) {
```
Comment (RichardDavies) [inline web/modules/custom/portland/portland.module:829]: It looks like there's an extra set of parentheses around the if statement. (Not really critical, but could be cleaned up a bit.)

File: web/modules/custom/portland/portland.module [hunk 2]
```diff
@@ -824,6 +824,17 @@ function _validate_document(&$form, \Drupal\Core\Form\FormStateInterface $form_s
...
+      $form_state->setErrorByName('field_document', t('Must have either an uploaded document or an eFiles link'));
+      return;
+    }
+    else if ( ( !empty($efiles_link[0]["uri"]) && !empty($document[0]["fids"]) ) ) {
```
Comment (RichardDavies) [inline web/modules/custom/portland/portland.module:833]: Again extra set of parentheses for the else if.

File: web/modules/custom/portland/portland.module [hunk 3]
```diff
@@ -1477,3 +1457,56 @@ function portland_form_user_form_alter(&$form, FormStateInterface $form_state) {
...
+/**
+ * Implements hook_media_presave().
+ */
+function portland_media_presave($media)
+{
+  // Get file size and mime type from eFiles link
+  if($media->hasField('field_efiles_link')) {
+    if (!empty($media->field_efiles_link) && !empty($media->field_efiles_link[0]->uri)) {
+      $efiles_link = $media->field_efiles_link[0]->uri;
+      $user_input_file_url = strtolower($efiles_link);
```
Comment (RichardDavies) [inline web/modules/custom/portland/portland.module:1470]: `$user_input_file_url` Appears to be unused in this function, so why not just assign this directly to `$external_file_url`?

**Review by RichardDavies — APPROVED** [review]
Looks good. I've approved the PR, although I did note a few trivial code clean up options. But I'll leave it up to you whether or not you want to do them or not.

File: web/modules/custom/portland/portland.module (hunk 1 above)
Comment (kxwang) [reply web/modules/custom/portland/portland.module]: Good catch! Extra parentheses removed.

File: web/modules/custom/portland/portland.module (hunk 2 above)
Comment (kxwang) [reply web/modules/custom/portland/portland.module]: Extra parentheses removed.

File: web/modules/custom/portland/portland.module (hunk 3 above)
Comment (kxwang) [reply web/modules/custom/portland/portland.module]: Removed the variable `$user_input_file_url`

**Review by RichardDavies — APPROVED** [review]

**Pull request merged**